coral/src/app/features/topics/schema-request/TopicSchemaRequest.test.tsx

```diff
@@ -130,20 +127,64 @@ describe("TopicSchemaRequest", () => {
       useQuerySpy.mockRestore();
     });
 
-    it("does not redirect user if env query is part of list of environments", async () => {
+    it("does not redirect user if env ID query is part of list of environments", async () => {
       mockGetSchemaRegistryEnvironments.mockResolvedValue([
         ...mockedGetSchemaRegistryEnvironments,
       ]);
 
       customRender(<TopicSchemaRequest topicName={testTopicName} />, {
         queryClient: true,
         memoryRouter: true,
-        customRoutePath: "/topic/testtopic/request-schema?env=999",
+        customRoutePath: "/topic/testtopic/request-schema?env=1",
+      });
+
+      const form = getForm();
+      expect(form).toBeVisible();
+
+      await waitForElementToBeRemoved(
+        screen.getByTestId("environments-select-loading")
+      );
+
+      const select = within(form).getByRole("combobox", {
+        name: /Environment/i,
+      });
+
+      expect(select).toBeDisabled();
+      expect(select).toHaveValue("1");
+
+      // only testing this would always return green - even waitFor will return always
+      // true, since the mock is not called directly, so waitFor will check, confirm
+      // it has not been called because it has not happened yet. Checking for the
+      // form makes implicitly sure that navigate was not called (otherwise no form)
+      // and this assertion is mostly for readability
+      expect(mockedUsedNavigate).not.toHaveBeenCalled();
+    });
+
+    it("does not redirect user if env name query is part of list of environments", async () => {
+      mockGetSchemaRegistryEnvironments.mockResolvedValue([
+        ...mockedGetSchemaRegistryEnvironments,
+      ]);
+
+      customRender(<TopicSchemaRequest topicName={testTopicName} />, {
+        queryClient: true,
+        memoryRouter: true,
+        customRoutePath: "/topic/testtopic/request-schema?env=INFRA",
       });
 
       const form = getForm();
       expect(form).toBeVisible();
 
+      await waitForElementToBeRemoved(
+        screen.getByTestId("environments-select-loading")
+      );
+
+      const select = within(form).getByRole("combobox", {
+        name: /Environment/i,
+      });
+
+      expect(select).toBeDisabled();
+      await waitFor(() => expect(select).toHaveValue("3"));
```

Comment:
Bit extra, but now reading the tests I would probably also add `.toHaveDisplayValue` check here. From reading the test I can see "value in query is INFRA, value in select is 3". To verify that 3 is in fact INFRA, one would have to check to data defined at the beginning. 

By adding an additional `toHaveDisplayValue("INFRA")`, it would be clear. 

And I would at it at the `it` block checking for id in query, too,  for consistency :D

Reply:
This makes sense to me ^^

Reply:
[6a37ac1](https://github.com/aiven/klaw/pull/1344/commits/6a37ac19435d92447c54d24df8f59df5501c8c14)